src/common/audit-logs/interceptors/audit-log.interceptor.spec.ts

```diff
@@ -0,0 +1,144 @@
+    });
+
+  const createCallHandler = (response: any): CallHandler => ({
+    handle: jest.fn().mockReturnValue(of(response)),
+  });
+
+  beforeEach(() => {
+    reflector = createMock<Reflector>();
+    auditLogService = createMock<AuditLogService>();
+    interceptor = new AuditLogInterceptor(reflector, auditLogService);
+  });
+
+  afterEach(() => {
+    jest.clearAllMocks();
+  });
+
+  it('When the interceptor is instantiated, then it should be defined', () => {
+    expect(interceptor).toBeDefined();
+  });
+
+  describe('intercept', () => {
+    it('When no audit log metadata is found, then it should pass through without logging', async () => {
+      const context = createContext({});
+      const next = createCallHandler({ data: 'test' });
+      reflector.get.mockReturnValue(undefined);
+
+      const result = await lastValueFrom(interceptor.intercept(context, next));
+
+      expect(result).toEqual({ data: 'test' });
+      expect(auditLogService.log).not.toHaveBeenCalled();
+    });
+
+    it('When audit log metadata is found with minimal config, then it should create audit log with defaults', async () => {
+      const metadata: AuditLogConfig = { action: AuditAction.TfaEnabled };
+      const request = { user: newUser() };
+      const context = createContext(request);
+      const next = createCallHandler({ success: true });
+
+      reflector.get.mockReturnValue(metadata);
+      auditLogService.log.mockResolvedValue(undefined);
+
+      await lastValueFrom(interceptor.intercept(context, next));
+
+      expect(auditLogService.log).toHaveBeenCalledWith({
+        action: AuditAction.TfaEnabled,
+        entityType: AuditEntityType.User,
+        entityId: request.user.uuid,
+        performerType: AuditPerformerType.User,
+        performerId: request.user.uuid,
+        metadata: undefined,
+      });
+    });
+
+    it('When using custom extractors, then it should extract values from paths and callbacks', async () => {
+      const metadata: AuditLogConfig = {
+        action: AuditAction.WorkspaceCreated,
+        entityType: AuditEntityType.Workspace,
+        entityId: (_req, res) => res.workspace.id,
+        performerId: 'body.creatorId',
+        performerType: AuditPerformerType.System,
+        metadata: ['body.name'],
+      };
+      const request = {
+        user: { uuid: newUser() },
+        body: { creatorId: 'creator-789', name: 'My Workspace' },
+      };
+      const response = { workspace: { id: 'workspace-456' } };
+      const context = createContext(request);
+      const next = createCallHandler(response);
+
+      reflector.get.mockReturnValue(metadata);
+      auditLogService.log.mockResolvedValue(undefined);
+
+      await lastValueFrom(interceptor.intercept(context, next));
+
+      expect(auditLogService.log).toHaveBeenCalledWith({
+        action: AuditAction.WorkspaceCreated,
+        entityType: AuditEntityType.Workspace,
+        entityId: 'workspace-456',
```

Comment:
Using `response.workspace.id` / `request.body.creatorId` / `request.body.name` to assert, instead of hardcoded values, improves the maintainability of this test